safe-global/safe-wallet-web
# Refactor: track tx origin

## What it solves

To measure how many txs are made through WalletConnect, we'll track the Safe App origin field in onchain/offchain interaction events.

I also refactored the emitted tx event bus to not pass around humanDescription and instead fetch it from CGW.

## How to test it

Non-Safe App txs:
1. Make a send token tx
2. Observe an "Onchain interaction" event tracked in the console
3. The "label" property of this event should be empty (indicating a tx initiated from our own UI)

Safe App txs:
1. Make a tx with any Safe App
2. Observe an "Onchain interaction" event tracked in the console
3. The label property should be populated with that Safe App's URL
